packages/caliper-core/test/worker/rate-control/recordRate.js

```diff
@@ -63,42 +74,262 @@ describe('RecordRate controller', () => {
+            sinon.assert.calledWith(fsAppendSyncStub.getCall(0), sinon.match.string, '100\n');
+            sinon.assert.calledWith(fsAppendSyncStub.getCall(1), sinon.match.string, '200\n');
+            sinon.assert.calledWith(fsAppendSyncStub.getCall(2), sinon.match.string, '300\n');
+
+
+            fsWriteSyncStub.restore();
+            fsAppendSyncStub.restore();
+        });
+
+        it('should export records to binary big endian format', async () => {
+            const msgContent = {
+                label: 'test',
+                rateControl: {
+                    type: 'record-rate',
+                    opts: {
+                        rateController: {
+                            type: 'zero-rate'
+                        },
+                        pathTemplate: '../tx_records_client<C>_round<R>.txt',
+                        outputFormat: 'BIN_BE'
+                    }
+                },
+                testRound: 0,  // Ensure roundIndex is set
+                txDuration: 250,
+                totalWorkers: 2
+            };
+            const testMessage = new TestMessage('test', [], msgContent);
+            const controller = RecordRate.createRateController(testMessage, stubStatsCollector, 0);
+            sinon.stub(controller.recordedRateController, 'end').resolves();
+
+            controller.records = [100, 200, 300];
+            const fsWriteSyncStub = sandbox.stub(fs, 'writeFileSync');
+
+            await controller.end();
+
+            sinon.assert.calledOnce(fsWriteSyncStub);
+            const buffer = fsWriteSyncStub.getCall(0).args[1];
+            buffer.readUInt32BE(0).should.equal(3);
+            buffer.readUInt32BE(4).should.equal(100);
+            buffer.readUInt32BE(8).should.equal(200);
+            buffer.readUInt32BE(12).should.equal(300);
+
+            fsWriteSyncStub.restore();
+        });
+
+        it('should export records to binary little endian format', async () => {
+            const msgContent = { ...defaultMsgContent };
+            msgContent.rateControl.opts.outputFormat = 'BIN_LE';
+            const testMessage = new TestMessage('test', [], msgContent);
+            const controller = RecordRate.createRateController(testMessage, stubStatsCollector, 0);
+
+            sandbox.stub(controller.recordedRateController, 'end').resolves();
+
+            controller.records = [100, 200, 300];
+            const fsWriteSyncStub = sandbox.stub(fs, 'writeFileSync');
+
+            await controller.end();
+
+            sinon.assert.calledOnce(fsWriteSyncStub);
+            const buffer = fsWriteSyncStub.getCall(0).args[1];
+            buffer.readUInt32LE(0).should.equal(3);
+            buffer.readUInt32LE(4).should.equal(100);
+            buffer.readUInt32LE(8).should.equal(200);
+            buffer.readUInt32LE(12).should.equal(300);
+
+            fsWriteSyncStub.restore();
+        });
+        it('should export to text format when output format is TEXT', async () => {
```

Comment:
This is already tested above. given that you track the writes to the file, we know it's only writing to text and not the others.